clientv3/txn.go

```diff
@@ -147,16 +148,16 @@ func (txn *txn) Commit() (*TxnResponse, error) {
 		}
 
 		if isHalted(txn.ctx, err) {
-			return nil, err
+			return nil, rpctypes.Error(err)
```

Comment:
`isHalted` seems outdated now that errors getting wrapped up in types. Maybe the `if isHalted` stuff can now be: `if herr := haltErr(txn.ctx, err) { return nil, herr }`

with something like this:

```go
func isHalted(ctx context.Context, err error) error {
        if strings.HasPrefix(grpc.ErrorDesc(err), "etcdserver: ") { return rpctypes.Error(err) }
        if cerr := ctx.Err(); cerr != nil { return cerr}
        return nil
}
```